### DISCUSSION TITLE
Getting Couldnt read file: open key.enc: no such file or directory while running for AWS KMS

### DISCUSSION THREAD

**Discussion by pavanreddy07385** [bruj0/vault-recovery-key#16] [General]
can u pls help us resolve this issue?

 ./vault-recovery-key  -enc-key key.enc -env awskms -shamir-shares 5 -shamir-threshold 3
INFO[0000] Starting version 0.2                         
INFO[0000] Starting with environment awskms             
FATA[0000] Couldnt read file: open key.enc: no such file or directory 

where do we get this key.enc from?

**Comment by jdfriedma** [comment]
@pavanreddy07385 you generate that key by running the operations [in this section](https://github.com/bruj0/vault-recovery-key#encrypted-recovery-keys-dump) against the relevant storage entries. Give that a shot and see if that works for you.

**Reply by pavanreddy07385** [reply]
@jdfriedma We are running vault in K8s, using Postgres as a back-end, can you pls provide steps to retrieve recovery keys.

**Reply by jdfriedma** [reply]
@pavanreddy07385 I am not sure that this utility will work with the Postgres storage engine. As @bruj0 said in the readme, this has only been tested with the file storage engine and the consul storage engine. You'd have to check to see if there's a corresponding entry in the Postgres database for the recovery keys and if it's in a format that can be unmarshalled into something this utility would work on.
